common/autoware_universe_utils/include/autoware/universe_utils/geometry/geometry.hpp

```diff
@@ -583,6 +584,121 @@ std::optional<geometry_msgs::msg::Point> intersect(
  */
 bool intersects_convex(const Polygon2d & convex_polygon1, const Polygon2d & convex_polygon2);
 
+// Alternatives for Boost.Geometry ----------------------------------------------------------------
+
+// TODO(mitukou1109): remove namespace when migrating to alternatives
+namespace alt
+{
+class Vector2d
+{
+public:
+  Vector2d() : x_(0.0), y_(0.0) {}
+
+  Vector2d(const double x, const double y) : x_(x), y_(y) {}
+
+  double cross(const Vector2d & other) const { return x_ * other.y() - y_ * other.x(); }
+
+  double dot(const Vector2d & other) const { return x_ * other.x() + y_ * other.y(); }
+
+  double norm() const { return std::hypot(x_, y_); }
+
+  Vector2d vector_triple(const Vector2d & v1, const Vector2d & v2) const
+  {
+    const auto tmp = this->cross(v1);
+    return {-v2.y() * tmp, v2.x() * tmp};
+  }
+
+  const double & x() const { return x_; }
+
+  double & x() { return x_; }
```

Comment:
Is this reference (&) effective?
If it's not so effective, my opinion is removing it so that the user of this function will not assign a value to `x()` by mistake.

Reply:
Ah, in that case, we have to create another setter function.

Reply:
We have two options.
- option1.
    - x, y are private variables
    - create getter/setter functions
- option2.
    - x, y are public variables

This clas is very simple, so I feel it's fine and better to choose option2. What do you think?

Reply:
Isn't it common for accessors to return references? I intend to use `double & x()` to assign values.

Reply:
My concern is as follows, but this may be my misunderstanding since `auto x` is not `auto & x` in this case.
```
auto x = vec.x();
x = 2.0; // vec.x() is overwritten.
```

Then, the thing is whether we need to make x and y private or not as we can access x, y, and z of geometry_msgs::msg::Point for example. I don't have a strong opinion here, so let's keep the current implementation. Thank you.